community/kernel/src/main/java/org/neo4j/kernel/api/AssertOpen.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.api;
+
+/**
+ * Used to verify that some source of operation is still open when operation is performed.
+ * E.g. verify that transaction is still open when lazily reading property values from store.
+ */
+public interface AssertOpen
+{
+    /**
+     * Assert that source tied to instance is open.
+     * Should throw exception with reason if source is not open.
+     */
+    void assertOpen();
```

Comment:
`KernelStatement` will through two different variants of `RuntimeException`.
As that is an unchecked exception I would not put it in method signature, but maybe clarify it in javadoc. Sounds good?

From `KernelStatement`:
```
    @Override
    public void assertOpen()
    {
        if ( referenceCount == 0 )
        {
            throw new NotInTransactionException( "The statement has been closed." );
        }

        Status terminationReason = transaction.getReasonIfTerminated();
        if ( terminationReason != null )
        {
            throw new TransactionTerminatedException( terminationReason );
        }
    }
```